version: '3.8'

services:
  mysql:
    image: mysql:latest
    container_name: mysql
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: cafe_shop
      MYSQL_USER: cafe_user
      MYSQL_PASSWORD: cafe_password
      TZ: Asia/Ho_Chi_Minh
    ports:
      - "3306:3306"
    volumes:
      - cf_db_data:/var/lib/mysql
      - ./sql:/docker-entrypoint-initdb.d
    networks:
      - cf-net
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 5s
      retries: 5

  authentication:
    image: authentication:latest
    container_name: authentication
    build:
      context: ./authentication
      dockerfile: ./Dockerfile
    ports:
      - "9090:9090"
    environment:
      - DB_HOST=mysql
      - DB_PORT=3306
      - DB_NAME=cafe_shop
      - DB_USERNAME=cafe_user
      - DB_PASSWORD=cafe_password
      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
    restart: always
    networks:
      - cf-net
    depends_on:
      - mysql

volumes:
  cf_db_data:

networks:
  cf-net:
    driver: bridge
    ipam:
      driver: default
